it/src/test/java/com/google/cloud/teleport/it/bigquery/DefaultBigQueryResourceManagerTest.java

```diff
@@ -0,0 +1,373 @@
+  private static final String TEST_ID = "test-id";
+  private static final String PROJECT_ID = "test-project";
+  private static final String REGION = "us-central1";
+
+  private DefaultBigQueryResourceManager testManager;
+
+  @Before
+  public void setUp() {
+    testManager =
+        DefaultBigQueryResourceManager.builder(TEST_ID, PROJECT_ID).setBigQuery(bigQuery).build();
+  }
+
+  private void prepareCreateDataset() {
+    when(bigQuery.create((DatasetInfo) any())).thenReturn(dataset);
+    Mockito.lenient().when(dataset.getDatasetId().getDataset()).thenReturn(DATASET_ID);
+  }
+
+  private void prepareCreateTable() {
+    when(dataset.get(any())).thenReturn(table);
+    when(bigQuery.getTable(any())).thenReturn(null);
+    Mockito.lenient().when(table.getTableId().getTable()).thenReturn(TABLE_NAME);
+  }
+
+  @Test
+  public void testBuilderSetsValuesCorrectly() {
+    DefaultBigQueryResourceManager.Builder tmBuilder =
+        DefaultBigQueryResourceManager.builder(TEST_ID, PROJECT_ID);
+    assertThat(tmBuilder.getDatasetId())
+        .matches(TEST_ID.replace('-', '_') + "_\\d{8}_\\d{6}_\\d{6}");
+    tmBuilder
+        .setProjectId("fake_project_id")
+        .setTestId("fake_test_id")
+        .setDatasetId(DATASET_ID)
+        .setCredentials(credentials)
+        .setBigQuery(bigQuery);
+
+    assertThat(tmBuilder.getCredentials()).isEqualTo(credentials);
+    assertThat(tmBuilder.getProjectId()).isEqualTo("fake_project_id");
+    assertThat(tmBuilder.getTestId()).isEqualTo("fake_test_id");
+    assertThat(tmBuilder.getDatasetId()).matches(DATASET_ID);
+    assertThat(tmBuilder.getBigQuery()).isEqualTo(bigQuery);
+
+    DefaultBigQueryResourceManager tm = tmBuilder.build();
+
+    assertThat(tm.getProjectId()).isEqualTo("fake_project_id");
+    assertThat(tm.getTestId()).isEqualTo("fake_test_id");
+    assertThat(tm.getDatasetId()).matches(DATASET_ID);
+  }
+
+  @Test
+  public void testGetProjectIdReturnsCorrectValue() {
+    assertThat(testManager.getProjectId()).isEqualTo(PROJECT_ID);
+  }
+
+  @Test
+  public void testGetDatasetIdReturnsCorrectValue() {
+    DefaultBigQueryResourceManager.Builder tmBuilder =
+        DefaultBigQueryResourceManager.builder(TEST_ID, PROJECT_ID);
+    tmBuilder.setDatasetId(DATASET_ID).setCredentials(credentials);
+    DefaultBigQueryResourceManager tm = tmBuilder.build();
+
+    assertThat(tm.getDatasetId()).isEqualTo(DATASET_ID);
+  }
+
+  @Test
+  public void testGetTestIdReturnsCorrectValue() {
+    assertThat(testManager.getTestId()).isEqualTo(TEST_ID);
+  }
+
+  @Test
+  public void testCreateDatasetShouldThrowErrorWhenDatasetCreateFails() {
+    when(bigQuery.create((DatasetInfo) any())).thenThrow(RuntimeException.class);
+
+    assertThrows(
+        BigQueryResourceManagerException.class, () -> testManager.createDataset(DATASET_ID));
+  }
+
+  @Test
+  public void testCreateDatasetShouldNotCreateDatasetWhenDatasetAlreadyExists() {
+    prepareCreateDataset();
```

Comment:
nitpick-ish, but prepareCreateDataset() is 2 lines only, could be inlined in methods to make it flow better and have more self-contained tests

Reply:
Fixed. I also did the same for prepareCreateTable.